From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch 02-09

  Description: ctcmpc: Unknown symbol reported for extern callback functions
  Symptom:     snaipx which is part of CSL is unable to load ctcmpc devices
  Problem:     EXPORT_SYMBOL was missing for the extern functions
  Solution:    add EXPORT_SYMBOL statements for each extern callback function
  Problem-ID:  14424
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/net/ctcmpc.h	18 Jan 2006 11:27:59 -0000	1.4
+++ linux-2.5/drivers/s390/net/ctcmpc.h	13 Apr 2006 02:41:45 -0000	1.4.2.1
@@ -41,5 +41,8 @@
 extern void ctc_mpc_establish_connectivity(int,callbacktypei3);
 extern void ctc_mpc_dealloc_ch(int);
 extern void ctc_mpc_flow_control(int,int);
-
+EXPORT_SYMBOL(ctc_mpc_alloc_channel);
+EXPORT_SYMBOL(ctc_mpc_establish_connectivity);
+EXPORT_SYMBOL(ctc_mpc_dealloc_ch);
+EXPORT_SYMBOL(ctc_mpc_flow_control);
 #endif

